[Reland][FSDP Optim State] Remove checkpoint prefix

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80936

Reland which adds __all__ to torch/utils/checkpoint to silence public binding errors.

rohan-varma committed:
Revert "Revert "[FSDP Optim State] Remove checkpoint prefix (#80480)""

This reverts commit fe361dede45747067e39669573cbcf48be4fe4f9.

[ghstack-poisoned]

rohan-varma committed:
Update on "Revert "Revert "[FSDP Optim State] Remove checkpoint prefix (#80480)"""


This reverts commit fe361dede45747067e39669573cbcf48be4fe4f9.

[ghstack-poisoned]

anjali411 commented on test/allowlist_for_publicAPI.json:
```diff
@@ -1960,6 +1960,14 @@
     "Serialization",
     "wrapper_singleton"
   ],
+  "torch.utils.checkpoint": [
```
I am not sure why this is being triggered in your PR ... but anyway this is not the right thing to do. I'd suggest adding an `__all__` in `torch/utils/checkpoint.py` if needed

rohan-varma committed:
Update on "Revert "Revert "[FSDP Optim State] Remove checkpoint prefix (#80480)"""


This reverts commit fe361dede45747067e39669573cbcf48be4fe4f9.

[ghstack-poisoned]

rohan-varma committed:
Update on "Revert "Revert "[FSDP Optim State] Remove checkpoint prefix (#80480)"""


This reverts commit fe361dede45747067e39669573cbcf48be4fe4f9.

[ghstack-poisoned]

rohan-varma committed:
Update on "Revert "Revert "[FSDP Optim State] Remove checkpoint prefix (#80480)"""


This reverts commit fe361dede45747067e39669573cbcf48be4fe4f9.

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge

malfet commented:
@rohan-varma please avoid `Revert "Revert` commits and at least add some explanation what have changed between this land attempt and the last one

rohan-varma commented:
@malfet Thanks for letting me know - I updated the PR title and added some info, though unfortunately I guess that won't reflect in the commit history.
